### PR TITLE
kraken.parseWSOrder string math

### PR DESCRIPTION
**Pull request by samgermain**
```
% kraken watchOrders
Debugger listening on ws://127.0.0.1:51700/f43bb4ea-7ea0-4cbc-802a-92cc52e80039
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
2023-01-13T17:00:26.345Z
Node.js: v18.4.0
CCXT v2.6.4
kraken.watchOrders ()
                 id | clientOrderId |     timestamp |                 datetime | lastTradeTimestamp | status |   symbol |  type | timeInForce | postOnly | side | price |  stopPrice | triggerPrice | cost | amount | filled | average | remaining |                          fee | trades |                           fees
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
OQQGF3-6VZYS-YKHE3V |             0 | 1673573001411 | 2023-01-13T01:23:21.411Z |                    |   open | ADA/USDT | limit |             |          |  buy |   0.3 | 0.00000000 |   0.00000000 |    0 |     15 |      0 |         |        15 | {"cost":0,"currency":"USDT"} |     [] | [{"cost":0,"currency":"USDT"}]
1 objects
                 id | clientOrderId | timestamp | datetime | lastTradeTimestamp |   status | symbol | type | timeInForce | postOnly | side | price | stopPrice | triggerPrice | cost | amount | filled | average | remaining | fee | trades | fees
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
OQQGF3-6VZYS-YKHE3V |             0 |           |          |                    | canceled |        |      |             |          |      |       |           |              |    0 |        |      0 |         |           |     |     [] |   []
1 objects
                 id | clientOrderId |     timestamp |                 datetime | lastTradeTimestamp | status |   symbol |   type | timeInForce | postOnly | side |    price |  stopPrice | triggerPrice |      cost | amount | filled |  average | remaining |                                   fee | trades |                                    fees
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
OALFYA-TH3YZ-FPVQRZ |             0 | 1673629310645 | 2023-01-13T17:01:50.645Z |                    | closed | ADA/USDT | market |         IOC |          |  buy | 0.330539 | 0.00000000 |   0.00000000 | 11.568865 |     35 |     35 | 0.330539 |         0 | {"cost":0.03007905,"currency":"USDT"} |     [] | [{"cost":0.03007905,"currency":"USDT"}]
1 objects
```

### CODE REVIEW

**Review by pcriadoperez — APPROVED** [review]
✅ LGTM
```python
kraken.watchOrders()
[{'amount': 0.01,
  'average': None,
  'clientOrderId': '0',
  'cost': 0.0,
  'datetime': '2022-11-04T00:45:05.757Z',
  'fee': {'cost': 0.0, 'currency': 'USDT', 'rate': None},
  'fees': [{'cost': 0.0, 'currency': 'USDT', 'rate': None}],
  'filled': 0.0,
  'id': 'OXTU4O-PW5O2-FHEPJT',
  'info': {'avg_price': '0.00000',
           'cost': '0.00000',
           'descr': {'close': None,
                     'leverage': None,
                     'order': 'sell 0.01000000 ETH/USDT @ limit 1900.00000',
                     'ordertype': 'limit',
                     'pair': 'ETH/USDT',
                     'price': '1900.00000',
                     'price2': '0.00000',
                     'type': 'sell'},
           'expiretm': None,
           'fee': '0.00000',
           'limitprice': '0.00000',
           'misc': '',
           'oflags': 'fciq',
           'opentm': '1667522705.757622',
           'refid': None,
           'starttm': None,
           'status': 'open',
           'stopprice': '0.00000',
           'timeinforce': 'GTC',
           'userref': 0,
           'vol': '0.01000000',
           'vol_exec': '0.00000000'},
  'lastTradeTimestamp': None,
  'postOnly': None,
  'price': 1900.0,
  'remaining': 0.01,
  'side': 'sell',
  'status': 'open',
  'stopPrice': '0.00000',
  'symbol': 'ETH/USDT',
  'timeInForce': None,
  'timestamp': 1667522705757,
  'trades': [],
  'triggerPrice': '0.00000',
  'type': 'limit'}]
  ```

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
p kraken watchOrders "LTC/USDT"                
Python v3.10.8
CCXT v2.6.4
kraken.watchOrders(LTC/USDT)
[{'amount': 0.1,
  'average': None,
  'clientOrderId': '0',
  'cost': 0.0,
  'datetime': '2023-01-14T15:22:36.757Z',
  'fee': {'cost': 0.0, 'currency': 'USDT', 'rate': None},
  'fees': [{'cost': 0.0, 'currency': 'USDT', 'rate': None}],
  'filled': 0.0,
  'id': 'OP3JX7-A4VGV-FESTBP',
  'info': {'avg_price': '0.000000',
           'cost': '0.000000',
           'descr': {'close': None,
                     'leverage': None,
                     'order': 'buy 0.10000000 LTC/USDT @ market 0.000000',
                     'ordertype': 'market',
                     'pair': 'LTC/USDT',
                     'price': '0.000000',
                     'price2': '0.000000',
                     'type': 'buy'},
           'expiretm': None,
           'fee': '0.000000',
           'limitprice': '0.000000',
           'misc': '',
           'oflags': 'fciq',
           'opentm': '1673709756.757828',
           'refid': None,
           'starttm': None,
           'status': 'pending',
           'stopprice': '0.000000',
           'timeinforce': 'GTC',
           'userref': 0,
           'vol': '0.10000000',
           'vol_exec': '0.00000000'},
  'lastTradeTimestamp': None,
  'postOnly': None,
  'price': None,
  'remaining': 0.1,
  'side': 'buy',
  'status': 'open',
  'stopPrice': 0.0,
  'symbol': 'LTC/USDT',
  'timeInForce': 'IOC',
  'timestamp': 1673709756757,
  'trades': [],
  'triggerPrice': 0.0,
  'type': 'market'}]
```

**Review by carlosmiei — APPROVED** [review]

File: js/pro/kraken.js [hunk 1]
```diff
@@ -1004,7 +1004,7 @@ module.exports = class kraken extends krakenRest {
         if (rawTrades !== undefined) {
             trades = this.parseTrades (rawTrades, market, undefined, undefined, { 'order': id });
         }
-        const stopPrice = this.safeString (order, 'stopprice');
+        const stopPrice = this.safeNumber (order, 'stopprice');
```
Comment (samgermain) [inline js/pro/kraken.js:1007]: I thought `safeOrder` expected unparsed strings?

Comment (carlosmiei) [reply js/pro/kraken.js:1007]: @samgermain it does for most values, `stopPrice` is not one of them yet so we were sending a stringified number to the user

**Pull request merged**